Graphe Highstock

Sboune opened the pull request:
## Mise en place d'un graphe HighStock

Génère un graphe pour les Capteurs sélectionné dans la vue 3D en fonction des dates données dans le formulaire

![Graphe](https://cloud.githubusercontent.com/assets/5512725/5991807/7da34714-aa07-11e4-9dc6-794e93727c82.png)

Les données sont chargé dynamiquement en Ajax lors des zoom.

![graphezoom](https://cloud.githubusercontent.com/assets/5512725/5991816/d82185fc-aa07-11e4-92a3-36c88c1d6dc8.png)

Joystikman committed:
Début génération de graphes

Début de la génération des graphes selon les sondes sélectionnées.
Add: Ajout d'un fichier generer_graph.php qui est appelé par le bouton
"Générer" de index.php. Permet d'afficher les données d'un enregistrer
par un capteur pendant un intervalle donnée.
Add: création de creation_graph.js qui créera les graphes selon les
capteurs sélectionnés (pas encore implémenté).
Change: index.php pour qu'il appel generer_graph.php et ajout d'une
sélection de sonde.

Commit:
Mise en place de Highcharts sur la page index.php, à la place de C3

Change: création et parametrage du graph Highcharts dans le fichier graph.js.
Change: mise a jour de la bibliotheque Highcharts.
- exporting.js
- highcharts.js
Change: ajout des quelque ligne necessaire pour importer la bibliotheque Highcharts sur index.php.

Joystikman committed:
Génération tableau de donnée pour graphe

Changed: Modification de index.php pour ajouter des checkbox de selection de capteurs
Changed: Modification de generer_graph.php. Pour des sondes données et un interval donné, permet d'obtenir un tableau de tableau des valeurs enregistrées par les capteurs

Commit:
Ajout de test de l'ajout de Series (nouvelles courbes) au graph

Change: ajout d'une boucle for qui crée 5 series de 5 valeurs random puis ajout des series au au chart.

Joystikman committed:
Adaptation format des données pour le graphe

Changed: Modification de generer_graph.php et de la fonction getData() pour quelle renvoie un liste de string au format générer par le graphe.

Commit:
Association sonde et graph

Change: modification du type du fichier graph, js devient php
Change: inclusion de graph.php dans index.php
Change: recuperation des donne des generer_graph.php pour tracer les courbe de valeur, graph.php

Commit:
Passage de HighCharts à HighStock

Add: ajout de la bibliotheque HighStock.js
Changed: importation de la biblio dans index.php
Changed: modification des parametres afin que le graphe s'applique bien, dans graph.js.
Changed: suppression de la mini chart, ajout des boutons (+style) et du zoom, sur le graphe.
fix #18

Sboune committed:
Ajout de l'ajax pour la gestion de la courbe

Ajout de l'ajax pour la generation et les zooms dans le graphe

Azurency commented on index.php:
```diff
@@ -43,21 +47,26 @@
                   <div class="twelve columns">
-                    <input class="button-primary u-full-width" type="submit" value="Génerer">
+                    Sondes à selectionner :
+                    <input type="checkbox" name='capteurs[]' value="A1-1"> A1-1
```
Pourquoi y a-t-il encore des checkbox ? La vue 3D ajoute un tableau qui s'appelle `capteursId[]`, ces checkbox sont normalement inutiles (j'espère) il faudrait les supprimer.

Azurency commented:
Il faudrait déplacer le fichier _`graph.php`_ pour qu'il ne soit pas dans le dossier `assets/js` mais dans `includes/scripts`.

Azurency commented on assets/js/graph.php:
```diff
@@ -0,0 +1,259 @@
+		tab.push((Math.random()*10)-5);
+	};
+	tempP.push(tab);
+};
```
Si ce code est inutile, il faudrait carrément le supprimer

Azurency commented on index.php:
```diff
@@ -69,7 +78,9 @@
             <div class="box-section">
+              <?php valider() ?> <!-- Appelle generer_graph.php-->
               <div id="chart"></div>
+              <div id="test"></div>
```
à quoi sert ce `<div id="test">` ? c'est pas très propre d'avoir un _test_ dans l'index.

Azurency commented on includes/scripts/generer_graph.php:
```diff
@@ -0,0 +1,151 @@
+	}
+
+//fonction qui prend en paramètre le nom d'un capteur et renvoie le capteur
+	function getCapteur($nom){
```
Le nom d'un capteur ? Le nom n'est pas obligatoirement unique !

Azurency commented:
Revoit ton message de commit aussi pour : 
* corriger *graph* en *graphe*
* fermer l'issue #18 
* voir ce qu'a fait @Joystikman sur les issues #3 #4 et #5, il avait commit quelque chose qui n'est pas inclu ici, peut être que tu n'utilise plus du tout ce qu'il a fait, si c'est le cas, il faut que tu vérifie si les issues sont toujours présente, et si non, les fermer.

Azurency commented:
Revoit ton message de commit aussi pour :
- corriger graph en graphe
- fermer l'issue #18
- voir ce qu'a fait @Joystikman sur les issues #3 #4 et #5, il avait commit quelque chose qui n'est pas inclu ici, peut être que tu n'utilise plus du tout ce qu'il a fait, si c'est le cas, il faut que tu vérifie si les issues sont toujours présente, et si non, les fermer.

Sboune committed:
Nettoyage du code et graphe par defaut

Changed: Nettoyage du code inutilisé
Changed: Le fichier graph.php a été déplacé dans 'includes/script'
Add: Ajout d'un graphe par defaut avec un Capteur par dispositif sur les 7
derniers jours. fixes #3 #4 #5

Azurency commented:
:+1:
